tests/common/common_vec.h

```diff
@@ -136,28 +136,246 @@ bool check_vector_get_count_get_size(cl::sycl::vec<vecType, 1> inputVec) {
   return true;
 }
 
+// match float values to expected integer value
+template <typename T1, typename T2>
+T2 float_map_match(T1 floats[], T2 vals[], int size, T1 src) {
+  for (int i = 0; i < size; ++i) {
+    if (floats[i] == src) {
+      return vals[i];
+    }
+  }
+  return T2{};
+}
+
+template <typename T>
+static constexpr bool is_unsigned_type_v =  std::is_unsigned_v<T> ||
+                                            std::is_same_v<T, cl::sycl::cl_uchar> ||
+                                            std::is_same_v<T, cl::sycl::cl_uint> ||
+                                            std::is_same_v<T, cl::sycl::cl_ulong> ||
+                                            std::is_same_v<T, cl::sycl::cl_ushort>;
+
+template <typename sourceType, typename targetType>
+static constexpr bool if_FP_to_non_FP_conv_v =
+    is_cl_float_type<sourceType>::value && !is_cl_float_type<targetType>::value;
+
+template <typename vecType, int N, typename convertType>
+cl::sycl::vec<convertType, N> convert_vec(cl::sycl::vec<vecType, N> inputVec) {
+  cl::sycl::vec<convertType, N> resVec;
+  for (size_t i = 0; i < N; ++i) {
+    vecType elem = getElement(inputVec, i);
+    setElement<convertType, N>(resVec, i, convertType(elem));
+  }
+  return resVec;
+}
+
+template <typename vecType, int N, typename convertType>
+cl::sycl::vec<convertType, N> rte(cl::sycl::vec<vecType, N> inputVec) {
+  if
+    constexpr(if_FP_to_non_FP_conv_v<vecType, convertType>) {
+      const int size = 8;
+      vecType floats[size] = {2.3f,  3.8f,  1.5f,  2.5f,
+                              -2.3f, -3.8f, -1.5f, -2.5f};
+      convertType vals[size] = {2, 4, 2, 2,
+                                static_cast<convertType>(-2),
+                                static_cast<convertType>(-4),
+                                static_cast<convertType>(-2),
+                                static_cast<convertType>(-2)};
+      cl::sycl::vec<convertType, N> resVec;
+      for (size_t i = 0; i < N; ++i) {
+        vecType elem = getElement(inputVec, i);
+        auto elemConvert = float_map_match(floats, vals, size, elem);
+        setElement<convertType, N>(resVec, i, elemConvert);
+      }
+      return resVec;
+    }
+  else {
+    return convert_vec<vecType, N, convertType>(inputVec);
+  }
```

Comment:
This file might need some formatting

Reply:
I don't think they're useless, but we might be getting closer to a formatting holy war :) I'd say that we just need to keep the formatting consistent, and clang-format should take care of that.

Reply:
But that formatting is result of applying clang-format to this code (including constexpr on the different line from if) should it be changed?

Reply:
Ah fine, if this is the output of `clang-format`. I just need to get used to this strange `if constexpr`.
By the way, someone should look some day at the top `.clang-format` to see how it fits with the recent `clang-format`.

Reply:
The formatting is definitely off here. 
Here is clang-format (trunk version) output: https://godbolt.org/z/8cM91KT3j

Reply:
I apologize, it was result of old version of `clang-format`. Updated with fixed formatting in 846a2bda1824821d3b295c69417eba7ff20da618

Reply:
> And also removing useless `{ }` like in the LLVM project?

To address this comment we can remove else-branch completely. if-branch ends with `return`, it safe to write like this:
```suggestion
  } // end of if-branch
  return convert_vec<vecType, N, convertType>(inputVec);
```

Reply:
Done.